### PR TITLE
ORC-418. Fix broken docker builds.

### PR DESCRIPTION
**Pull request by omalley**
Ok this patch fixes a lot of the compatibility problems that have been added recently:

- use of noexcept instead of ORC_NOEXCEPT
- replace use of nullptr with ORC_NULLPTR
- replace use of override with ORC_OVERRIDE
- testing for problematic cases of std::to_string have incomplete type coverage
- minor field initialization changes to make the older g++ happy
- make all of the docker scripts install openjdk 8
- the very old linux versions need an older version of protobuf to compile correctly
- the old linux versions need an older version of snappy to compile correctly
- modify the run-all.sh script to pass the correct options to cmake
- fix some of the new tests to not depend on the default charset
- fix one case where an old g++ didn't like initializing a vector with a literal

### CODE REVIEW

File: c++/src/RLEv1.hh [hunk 1]
```diff
@@ -22,6 +22,7 @@
 #include "Adaptor.hh"
 #include "RLE.hh"
 
+#include <cstdint>
```
[the remark this answers is not in the document]
Comment (omalley) [reply c++/src/RLEv1.hh]: Oh, thank you. I couldn't believe that uint wasn't built in. I couldn't figure out what to google to find the answer and I don't have a windows box.

**Comment by omalley** [comment]
Ok, that didn't work well either. I'm going to switch it to uint64_t, which works in other contexts.

**Comment by omalley** [comment]
The final patch makes some minor tweaks to the run-all script, so that the output is saved to a file rather than stdout.

File: docker/centos6/Dockerfile [hunk 2]
```diff
@@ -42,6 +43,22 @@ WORKDIR /root
 RUN wget "https://www.apache.org/dyn/closer.lua?action=download&filename=/maven/maven-3/3.3.9/binaries/apache-maven-3.3.9-bin.tar.gz" -O maven.tgz
 RUN tar xzf maven.tgz
 RUN ln -s /root/apache-maven-3.3.9/bin/mvn /usr/bin/mvn
+# install a local build of protobuf
+RUN wget "https://github.com/protocolbuffers/protobuf/archive/v2.5.0.tar.gz" \
```
Comment (majetideepak) [inline docker/centos6/Dockerfile]: Does it make sense to make these version numbers as parameters?

**Comment by majetideepak** [comment]
Rest of the changes look straightforward to me.
+1

**Pull request closed without merging**